framework/utils/CFormatter.php

```diff
@@ -235,7 +235,7 @@ protected function normalizeDateValue($time)
 				return strtotime($time);
 		}
 		elseif (class_exists('DateTime', false) && $time instanceof DateTime)
-			return $time->getTimestamp();
+			return $time->getTimestamp() + $time->getOffset();
```

Comment:
A Unix timestamp is always UTC, there is no point in adding an offset here.

I see however that there may be a problem with this code in general so you might open an issue describing the problem you see when using the formatter.